[docs] add more explanations in how to prepare for `babs-init`

zhao-cy opened the pull request.

tsalo commented on docs/source/babs-init.rst:
```diff
@@ -20,6 +20,11 @@ Command-Line Arguments
       :ref:`how-to-define-name-of-input-dataset` below for general guidelines
       and specific restrictions.
 
+   --keep-if-failed : @after
```
Can you try this?

```suggestion
   --keep_if_failed, --keep-if-failed : @after
```

tsalo replied:
Darn, sorry.

zhao-cy replied:
That's okay, no worries! If you don't have other suggestions I might just add those words into CLI - won't matter a ton.

zhao-cy commented:
In this PR, I've changed:

- added more explanations in docs/rephrased previous docs:
  - docs for how to prepare a container datalad dataset for BABS
  - docs for how to prepare container configuration yaml file for BABS
  - docs for what if `babs-init` fails

zhao-cy commented:
@djarecka I'm merging this branch to main now. If you check the updated readthedocs in a minute, you'll see more explanations on:
- [how to prepare a container datalad dataset](https://pennlinc-babs.readthedocs.io/en/latest/preparation_container.html) (which I've gone thru with you last week);
- [how to prepare container configuration YAML file](https://pennlinc-babs.readthedocs.io/en/latest/preparation_config_yaml_file.html)

Let me know if any of these docs does is hard to understand/does not make sense!
